### DISCUSSION TITLE
How to limit the maxWidth of textbox？

### DISCUSSION THREAD

**Discussion by Jiiiin** [fabricjs/fabric.js#7891] [Q&A] [answered]
I have a textbox with determined width and height，want to limit the input text to not exceed the set width, so I will give an error prompt and give up the content entered this time.

I have tried whether I can monitor text changes, or size changes, and handle them in events, but I haven't found a suitable event.

https://codesandbox.io/s/still-haze-zys42f?file=/src/App.tsx

**Comment by ShaMan123** [comment, marked as answer]
This is the method that changes the textbox width
https://github.com/fabricjs/fabric.js/blob/cf59e4518fd68fde00459c7315d84aab4a77d3cd/src/shapes/textbox.class.js#L322


Here you can override and check what you need and even return a different value:
https://github.com/fabricjs/fabric.js/blob/cf59e4518fd68fde00459c7315d84aab4a77d3cd/src/shapes/textbox.class.js#L292

**Reply by ShaMan123** [reply]
https://codesandbox.io/s/reverent-poitras-kf9ymy?file=/src/App.tsx

**Reply by ShaMan123** [reply]
I have overridden `_renderChars` as well. Take a look. Cursor should be handled as well.

**Reply by Jiiiin** [reply]
Thank you so much, this has inspired me a lot

**Reply by asturur** [reply]
I think the `changed` event should allow you to verify text widh and rollback the change in case, no?

**Reply by ShaMan123** [reply]
AFAIK there isn't a simple way of controlling how textbox performs layout.

**Comment by ShaMan123** [comment]
#8470

**Reply by GoldenFlash0927** [reply]
How is the height limit ?

**Reply by ShaMan123** [reply]
N/A 
You can look at the PR and try to impl
